src/dbwrapper.cpp

```diff
@@ -126,12 +126,26 @@ CDBWrapper::CDBWrapper(const fs::path& path, size_t nCacheSize, bool fMemory, bo
     : m_name{path.stem().string()}
 {
     penv = nullptr;
-    readoptions.verify_checksums = true;
-    iteroptions.verify_checksums = true;
     iteroptions.fill_cache = false;
     syncoptions.sync = true;
     options = GetOptions(nCacheSize);
     options.create_if_missing = true;
+
+    auto leveldbchecksum = gArgs.GetArg("-leveldbchecksum", "auto");
+    if(leveldbchecksum == "false"){
+        // set to true by default for all versions above 1.16 of LevelDB in GetOptions()
+        options.paranoid_checks = false;
+    }
+    else if (leveldbchecksum == "true") {
+        readoptions.verify_checksums = true;
+        iteroptions.verify_checksums = true;
+    }
+    else if (leveldbchecksum == "auto"){
```

Comment:
Should we remove the if statement here to make this the default else in case someone made a typo when setting leveldbchecksum?

Reply:
you are right we need to handle wrong input, what about a final else throwing an error or at least showing/logging a message and then set to default behaviour?